sentinelsat/sentinel.py

```diff
@@ -806,30 +885,43 @@ def _md5_compare(self, file_path, checksum, block_size=2 ** 13):
-        headers = {}
-        continuing = exists(path)
-        if continuing:
-            already_downloaded_bytes = getsize(path)
-            headers = {'Range': 'bytes={}-'.format(already_downloaded_bytes)}
-        else:
-            already_downloaded_bytes = 0
-        downloaded_bytes = 0
-        with closing(session.get(url, stream=True, auth=session.auth,
-                                 headers=headers, timeout=self.timeout)) as r, \
-                closing(self._tqdm(desc="Downloading", total=file_size, unit="B",
-                                   unit_scale=True, initial=already_downloaded_bytes)) as progress:
-            _check_scihub_response(r, test_json=False)
-            chunk_size = 2 ** 20  # download in 1 MB chunks
-            mode = 'ab' if continuing else 'wb'
-            with open(path, mode) as f:
-                for chunk in r.iter_content(chunk_size=chunk_size):
-                    if chunk:  # filter out keep-alive new chunks
-                        f.write(chunk)
-                        progress.update(len(chunk))
-                        downloaded_bytes += len(chunk)
-            # Return the number of bytes downloaded
-            return downloaded_bytes
+    def _download(self, url, path, session, file_size, product_info=None):
+        try:
+            headers = {}
+            continuing = exists(path)
+            if continuing:
+                already_downloaded_bytes = getsize(path)
+                headers = {'Range': 'bytes={}-'.format(already_downloaded_bytes)}
+            else:
+                already_downloaded_bytes = 0
+            downloaded_bytes = 0
+            with closing(session.get(url, stream=True, auth=session.auth,
+                                     headers=headers, timeout=self.timeout)) as r, \
+                    closing(self._tqdm(desc="Downloading", total=file_size, unit="B",
+                                       unit_scale=True, initial=already_downloaded_bytes)) as progress:
+                _check_scihub_response(r, test_json=False)
+                chunk_size = 2 ** 20  # download in 1 MB chunks
+                mode = 'ab' if continuing else 'wb'
+                with open(path, mode) as f:
+                    for chunk in r.iter_content(chunk_size=chunk_size):
+                        if chunk:  # filter out keep-alive new chunks
+                            f.write(chunk)
+                            progress.update(len(chunk))
+                            downloaded_bytes += len(chunk)
+                if product_info is not None and product_info['title'] in self.failed_list_title:
+                    self.failed_list_title.remove(product_info['title'])
+                    self.failed_list_id.remove(product_info['id'])
+
+                # Return the number of bytes downloaded
+                return downloaded_bytes
+        # meant for catching other types of exceptions such as 500 internal server side errors
+        except Exception as ex:
```

Comment:
Same as above. Never catch-all for a whole body of code. What if you introduced a typo in the code? You would not be able to tell whether the download failed because of SciHub / a cosmic ray flipping a bit or because your code broke...

Reply:
As I explained above, there are two major issues with Fin-hub servers, so to test their server I had to catch the error raised by _download method. there was no need to do that if the server had not raised 500 internal errors! right now your library crashes using Fin-hub servers due to these issues. They have promised to solve this hopefully in this September.